src/proxy/util.rs

```diff
@@ -12,7 +12,12 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use crate::proxy::ProxyInputs;
+use drain::Watch;
```

Comment:
If we are going to do this abstraction + wrapper move (which I do like), I would suggest we take the opportunity kick `drain` to the curb, it's redundant wrt the drain functionality we already get with `tokio`, which was one of the points in https://github.com/istio/ztunnel/issues/899

Reply:
I partially did this by making our own simpler/more clearly named `drain` crate. but it is implemented in terms of the existing `drain` crate we use, since I don't see a problem with their implementation. LMK what you think

Reply:
It also makes it trivial to swap out if there is a better impl

Reply:
Yep yep, it's a definite improvement, we need our own drain crate either way. I don't have a problem with their impl I just think it's strictly redundant with/easily replaced by a handful of `tokio::sync` primitives for our uses, and we don't need it.

But that's wishlisting, not PR review.